src/common/wasm_util.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/common/wasm_util.h"
+#include <cstdio>
+#include <string.h>
+
+namespace proxy_wasm {
+namespace common {
+
+bool WasmUtil::getCustomSection(const char *begin, const char *end, std::string_view name,
+                                std::string_view &ret) {
+  // Skip the Wasm header.
```

Comment:
Since those are standalone functions, I think you should validate Wasm header here.

Reply:
we already validate Wasm header at the beginning of `load` so I created another checkWasmHeader util and use it there. Do you think we should invoke `checkWasmHeader` again here?

Reply:
changed to call checkWasmHeader on each function. Thanks for suggestion!